core/src/main/java/io/keen/client/java/KeenClient.java

```diff
@@ -1549,13 +1565,19 @@ private void handleSuccess(KeenCallback callback) {
      * @param keenProperties  A Map that consists of key/value pairs to override default properties.
      *                        ex: "timestamp" -&gt; Calendar.getInstance()
      */
-    private void handleSuccess(KeenCallback callback, KeenProject project, String eventCollection, Map<String, Object> event,
-			Map<String, Object> keenProperties) {
-    	handleSuccess(callback);
+    private void handleSuccess(KeenCallback callback,
+                               KeenProject project,
+                               String eventCollection,
+                               Map<String, Object> event,
+                               Map<String, Object> keenProperties) {
+        handleSuccess(callback);
         if (callback != null) {
             try {
                 if(callback instanceof KeenDetailedCallback){
-                	((KeenDetailedCallback)callback).onSuccess(project, eventCollection, event, keenProperties);
+                    ((KeenDetailedCallback)callback).onSuccess(project,
+                                                               eventCollection,
+                                                               event,
+                                                               keenProperties);
                 }
             } catch (Exception userException) {
                 // Do nothing.
```

Comment:
Bummer, I typed up an entire response to this and apparently never clicked submit. Anyway the idea is a telemetry module should *not* take down your app since it's not crucial for even reporting to work in order for the app to work. So, failures are explicitly handled in a callback, or else they're swallowed. I think for event write, this is the proper technique, but for queries I would not do this, because one is asking for data that presumably is needed to show the user something, so failure isn't a tolerable scenario. Especially, in this case, we shouldn't re-throw because it's in the callback to client code. If client code wants to throw an exception in their callback, well, OK, I guess we could opt to never prevent you from shooting yourself in the foot, but I'd say this is a better solution in an event reporting SDK. That said, we should probably log it, so I'll make a note to add logging to these.